prov/util/src/util_mem_monitor.c

```diff
@@ -208,6 +228,9 @@ int ofi_monitors_add_cache(struct ofi_mem_monitor **monitors,
 	for (iface = FI_HMEM_SYSTEM; iface < OFI_HMEM_MAX; iface++) {
 		cache->monitors[iface] = NULL;
 
+		if (!iface_enabled[iface])
```

Comment:
`hmem_ops[iface].initialized` gives you this exact information. We don't need the CFLAG checks in this function.

Reply:
thanks! I did not know that. That would make things much easier.

Reply:
It might not be in scope and you might have to move definitions around, but it is worthwhile given the cache is now HMEM-aware.

Reply:
In new revision, I introduce new utility function `ofi_hmem_is_interface_initialized(iface)`, to get that information.